File: Modules/Common/Resources/SIE.Web.Resources/WipResources/Commands/WipResourceRefreshCommand.cs
using SIE.Resources.WipResources;
using SIE.Web.Command;

namespace SIE.Web.Resources.WipResources.Commands
{
    /// <summary>
    /// 生产资源刷新命令
    /// </summary>
    [JsCommand("SIE.Web.Resources.WipResources.Commands.WipResourceRefreshCommand")]
    public class WipResourceRefreshCommand : ViewCommand
    {
        /// <summary>
        /// 命令执行逻辑
        /// </summary>
        /// <param name="args">参数</param>
        /// <param name="scope">作用域</param>
        /// <returns>错误信息字符串</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            return RT.Service.Resolve<WipResourceController>().RunSync();
        }
    }
}

File: Modules/Common/Resources/SIE.Web.Resources/WipResources/Commands/WipResourceRefreshCommand.js
SIE.defineCommand('SIE.Web.Resources.WipResources.Commands.WipResourceRefreshCommand', {
    meta: { text: "同步资源", group: "business", iconCls: "icon-ListConfig icon-blue" },
    execute: function (view, source) {
        SIE.Msg.wait("正在同步资源，请稍等...".t());
        view.execute({
            data: {},
            success: function (res) {
                if (res.Result)
                    SIE.Msg.showMessage(Ext.String.format('同步出错:{0}', res.Result));
                else
                    SIE.Msg.showMessage('同步成功'.L10N());
            }
        });
    }
});